spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -203,10 +210,29 @@ private Filter getFilter(Object[] parameters) {
 								+ this.queryMethod.getName());
 			}
 		}).toArray(Filter[]::new);
-		return filters.length > 1
+
+		builder.setFilter(
+				filters.length > 1
 				? CompositeFilter.and(filters[0],
 						Arrays.copyOfRange(filters, 1, filters.length))
-				: filters[0];
+				: filters[0]);
 	}
 
+	private <A> A ifCountExistsQuery(Supplier<A> ifCount, Supplier<A> ifExists,
```

Comment:
This is an odd code style. Why not have a simple method `boolean isCountOrExistsQuery`?

Reply:
Then you can't do the 3-case with count, exists, or neither separately.